src/components/ec/cuda/ec_cuda.c

```diff
@@ -319,6 +325,18 @@ static ucc_status_t ucc_ec_cuda_init(const ucc_ec_params_t *ec_params)
             return UCC_ERR_NOT_SUPPORTED;
         }
     }
+
+    if (cfg->use_cooperative_launch == 1) {
+        cudaDeviceGetAttribute(&supportsCoopLaunch,
+                               cudaDevAttrCooperativeLaunch, device);
+        if (!supportsCoopLaunch) {
+            cfg->use_cooperative_launch = 0;
+            ec_debug(&ucc_ec_cuda.super,
```

Comment:
I would change ec_debug to ec_warn, because we want to let user know that default cooperative group mode is not working